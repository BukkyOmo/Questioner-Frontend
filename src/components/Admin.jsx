import React, { Component } from 'react';
import { Link } from 'react-router-dom';

const image = 'https://res.cloudinary.com/forsetti/image/upload/v1558483538/Questioner/event4.jpg';

const AdminPage = () => (
  <React.Fragment>
    <div className='container'>
      <button
        className='admin-btn'
        type='button'
      >
        {' '}
        <Link to='/admin-create'><i className='fas fa-plus fa-lg' /></Link>
        {' '}
      </button>
      <main className='flex-admin'>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i className='far fa-edit fa-lg' />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt fa-lg'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-edit'
                />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-edit'
                />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i className='far fa-edit fa-lg' />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt fa-lg'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-edit'
                />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-edit'
                />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i className='far fa-edit fa-lg' />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt fa-lg'
                />
              </button>
            </div>
          </div>
        </div>
        <div className='admin-div'>
          <img src={image} alt='eventimage' className='admin-image' />
          <div>
            <h4>Andela Boot camp</h4>
            <div className='admin-buttons'>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-edit'
                />
              </button>
              <button
                className='inner-admin'
                type='button'
              >
                <i
                  className='far fa-trash-alt'
                />
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </React.Fragment>

);

export default AdminPage;
