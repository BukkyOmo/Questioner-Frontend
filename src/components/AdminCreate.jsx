// import React, { Component } from 'react';

// class AdminCreate extends Component {
//   constructor(props) {
//     super(props);
//     // this.state = {
//     //   topic: '',
//     //   location: '',
//     //   meetup_date: '',
//     //   image: '',
//     //   tags: ''
//     // };
//   }

//   // handleSubmit = () => {
//   //   console.log(this.state);
//   // }

//   render() {
//     const {
//       topic,
//       location,
//       meetupDate,
//       meetupImage,
//       meetupTag
//     } = this.state;
//     return (
//       <React.Fragment>
//         <form action='POST' className='flex_admincreate' onSubmit='this.handleSubmit'>
//           <div className='items'>
//             <h1>Add Meetup</h1>
//           </div>
//           <div>
//             <label Htmlfor='meetup_topic'>
//               Topic
//               <div className='items'>
//                 <input
//                   type='text'
//                   name='meetup_topic'
//                   id='meetup_topic'
//                   placeholder='topic'
//                   value={topic}
//                 />
//               </div>
//             </label>
//           </div>
//           <div>
//             <label Htmlfor='meetup_location'>
//               Location
//               <div className='items'>
//                 <input
//                   type='text'
//                   name='meetup_location'
//                   id='meetup_location'
//                   placeholder='location'
//                   value={location}
//                 />
//               </div>
//             </label>
//           </div>
//           <div>
//             <label Htmlfor='meetup_date'>
//               Date
//               <div className='items'>
//                 <input
//                   type='date'
//                   name='meetup_date'
//                   id='meetup_date'
//                   placeholder='date'
//                   value={meetupDate}
//                 />
//               </div>
//             </label>
//           </div>
//           <div>
//             <label Htmlfor='image'>
//               Image
//               <div className='items_create'>
//                 <input
//                   type='file'
//                   name='meetup_image'
//                   id='meetup_image'
//                   placeholder='image'
//                   value={meetupImage}
//                 />
//               </div>
//             </label>
//           </div>
//           <div>
//             <label Htmlfor='meetup_tag'>
//               Tags
//               <div className='items'>
//                 <input
//                   type='text'
//                   name='meetup_tag'
//                   id='meetup_tag'
//                   placeholder='Add tags'
//                   value={meetupTag}
//                 />
//               </div>
//             </label>
//           </div>
//           <div className='items'>

//             <input type='button' value='Add meetup' />
//           </div>
//         </form>
//       </React.Fragment>
//     );
//   }
// }

// export default AdminCreate;
