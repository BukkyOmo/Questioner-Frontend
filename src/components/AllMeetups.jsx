import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getAllMeetups } from '../actions';
import meetup1 from '../assets/images/event1.jpg';

export class AllMeetups extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.meetups();
  }

  meetups = () => {
    const { dispatch } = this.props;
    const token = localStorage.getItem('token');
    const config = {
      headers: {
        token
      }
    };
    dispatch(getAllMeetups(config));
  };

  render() {
    const { meetups } = this.props;
    return (
      <React.Fragment>
        <h1 className='meetups-header'>All Meetups</h1>
        <main className='main'>
          {meetups.map(meetup => (
            <div className='flex3' key={meetup.id}>
              <Link to='single-meetup.html'>
                <img src={meetup.image || meetup1} alt='eventimage' />
              </Link>
              <div className='flex-item'>
                <h3>{meetup.topic}</h3>

                <p>{meetup.happeningon}</p>

                <p>{meetup.location}</p>

                <p className='flex-item2'>
                  <Link to={`/meetups/${meetup.id}`}>View Details</Link>
                </p>
              </div>
            </div>
          ))}
        </main>

        <div className='center'>
          <div className='pagination'>
            <Link to='/'>&laquo;</Link>
            <Link to='/' className='active'>
              1
            </Link>
            <Link to='/'>2</Link>
            <Link to='/'>3</Link>
            <Link to='/'>4</Link>
            <Link to='/'>&raquo;</Link>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = ({ meetups }) => ({
  loading: meetups.isLoading,
  meetups: meetups.getMeetups
});

export default connect(mapStateToProps)(AllMeetups);
