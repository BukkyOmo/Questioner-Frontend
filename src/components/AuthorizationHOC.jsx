import React from 'react';
import { Route, Redirect } from 'react-router-dom';

import tokenExpired from '../utils/tokenExpired';

const AuthorizationHOC = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={() => {
      const token = localStorage.getItem('token');
      if (token && !tokenExpired(token)) {
        return <Component {...rest} />;
      }
      return <Redirect to='/signin' />;
    }}
  />
);

export default AuthorizationHOC;
