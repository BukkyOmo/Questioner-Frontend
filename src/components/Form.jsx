import React, { Component } from 'react';
import axios from 'axios';

class FormComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      username: '',
      password: '',
      cpassword: ''
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      email, username, password, cpassword
    } = this.state;
    const data = {
      email,
      username,
      password,
      cpassword
    };
    axios.post('https://bukkyomo-questioner.herokuapp.com/api/v1/auth/signup', { data })
      .then((res) => {
        console.log(res);
        console.log(res.data);
      });
    console.log(this.state);
  }

  handleChange = (e) => {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value });
    console.log(e.target.value);
  }

  render() {
    const {
      email, username, password, cpassword
    } = this.state;
    return (
      <React.Fragment>
        <main>
          <form method='POST' className='flex' onSubmit={this.handleSubmit}>
            <div className='items'>
              <h1>Sign Up</h1>
            </div>
            <div id='msg' />
            <div>
              <input
                type='email'
                id='email'
                placeholder='Email'
                name='email'
                onChange={this.handleChange}
                value={email}
              />
            </div>

            <div>
              <input
                type='text'
                id='username'
                placeholder='User Name'
                name='username'
                onChange={this.handleChange}
                value={username}
              />
            </div>

            <div>
              <input
                type='password'
                id='password'
                placeholder='Password'
                name='password'
                onChange={this.handleChange}
                value={password}
              />
            </div>

            <div>
              <input
                type='password'
                id='cpassword'
                placeholder='Confirm Password'
                name='cpassword'
                onChange={this.handleChange}
                value={cpassword}
              />
            </div>

            <div className='items'>
              <button
                className='button'
                id='signup'
                type='submit'
              >
                Signup
              </button>
            </div>

            <div className='items'>
              <p id='login'>
                Already have an account?
                <a href='/login'>
                  Sign In
                </a>
              </p>
            </div>
          </form>
        </main>
      </React.Fragment>
    );
  }
}

export default FormComponent;
