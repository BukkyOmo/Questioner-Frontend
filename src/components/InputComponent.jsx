import React from 'react';
import { string, bool, func } from 'prop-types';

/**
 * @description reusable input component
 * @returns {HTMLInputElement} input
 */
const InputElement = ({
  customClass,
  inputId,
  inputName,
  inputPlaceholder,
  inputValue,
  inputType,
  onChange,
  errorMessage,
  isRequired,
  isDisabled
}) => (
  <>
    <input
      className={`input-field ${customClass && customClass}`}
      id={inputId}
      name={inputName}
      placeholder={inputPlaceholder}
      value={inputValue}
      type={inputType}
      onChange={onChange}
      required={isRequired}
      disabled={isDisabled}
    />
    {errorMessage && <p className='form-error red-cl'>{errorMessage}</p>}
  </>
);
InputElement.propTypes = {
  /**
   * The input's value
   */
  inputValue: string,
  /**
   * The custom class containing extra styles for the input
   */
  customClass: string,
  /**
   * The id of the input
   */
  inputId: string,
  /**
   * The name attribute for the input
   */
  inputName: string,
  /**
   * The type attribute for the input
   */
  inputType: string,
  /**
   * The placeholder for the input
   */
  inputPlaceholder: string,
  /**
   * The function handler for a change event on the input
   */
  onChange: func.isRequired,
  /**
   * The error message to display for the input field
   */
  errorMessage: string,
  /**
   * if the input field is required or not
   */
  isRequired: bool,
  /**
   * if the input field is disabled or not
   */
  isDisabled: bool
};

InputElement.defaultProps = {
  customClass: '',
  inputId: '',
  inputName: '',
  inputType: '',
  inputPlaceholder: '',
  inputValue: '',
  errorMessage: '',
  isRequired: false,
  isDisabled: false
};

export default InputElement;
