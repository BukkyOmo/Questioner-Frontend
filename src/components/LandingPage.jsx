import React, { Component } from 'react';
import conversation from '../assets/images/conversation.svg';
import checklist from '../assets/images/checklist.svg';
import click from '../assets/images/click.svg';

const LandingPage = () => {
  return (
    <React.Fragment>
      <div className="banner">
        <h1 className="flex1">Welcome to Questioner</h1>
        <h2 className="flex1">Ask Questions about Upcoming Events.</h2>
      </div>
      <div className="banner2">
        <h2>Planning a meetup  and would love to get Questions upfront?</h2>
        <button className="flex2"><a href="signup.html">Get Started >></a></button>
      </div>

      <div className="center">
        <h1>How it works</h1>
      </div>
      <div class="svg">
       <div className="svg-item">

        <div className="justify-center">
            <img src={ conversation } alt="conversation-image" />
        </div>

        <p>Got any questions concerning any</p>
        <p>of the events posted here?</p>

      </div>
      <div className="svg-item">
        <div className="justify-center">
            <img src={ click } alt="checklist-image" />
        </div>
                <p>Click on the get Started button</p>
      </div>
      <div className="svg-item">
          <div className="justify-center">
              <img src={ checklist } alt="crowd-image" />
          </div>
            <p>View the event, ask your questions and vote</p>
            <p>for questions you also wish to know about</p>
        </div>
      </div>
    </React.Fragment>
    )
  }

export default LandingPage;
