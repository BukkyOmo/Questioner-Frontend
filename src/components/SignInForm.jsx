import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// import axios from 'axios';

class SignInForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { email, password } = this.state;
    const user = {
      email,
      password
    };

    // const data = await axios.post(endPoint, user);
    // this.setState(data);
  }

   handleChange = (e) => {
     this.setState({ [e.target.name]: e.target.value });
   }

   render() {
     const { email, password } = this.state;
     return (
       <React.Fragment>
         <main>
           <form className='flex' onSubmit={this.handleSubmit}>
             <div className='items'>
               <h1>Sign In</h1>
             </div>
             <div className='items'>
               <input
                 type='email'
                 name='email'
                 id='email'
                 placeholder='email'
                 onChange={this.handleChange}
                 value={email}
               />
             </div>
             <div className='items'>
               <input
                 type='password'
                 name='password'
                 id='password'
                 placeholder='Password'
                 onChange={this.handleChange}
                 value={password}
               />
             </div>
             <div className='items'>
               <button
                 id='login'
                 className='button'
                 type='submit'
               >
                  Sign In
               </button>
             </div>
             <div className='items'>
               <p>
                Dont have an account?
                 {' '}
                 <Link to='/signup'> Sign Up </Link>
               </p>
             </div>
             <div className='items forgot'>
               <p><a href='forgotpassword.html'> Forgot Password? </a></p>
             </div>
           </form>
         </main>
       </React.Fragment>
     );
   }
}

export default SignInForm;
