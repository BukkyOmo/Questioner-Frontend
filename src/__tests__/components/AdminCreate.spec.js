import React from "react";
import { shallow } from "enzyme";
import { CreateMeetup } from '../../components/AdminCreate';

describe('<CreateMeetup />', () => {
  const dispatch = jest.fn();
  const createmeetup = shallow(<CreateMeetup dispatch={dispatch} />);
  it('renders without crashing', () => {
     expect(createmeetup.find('div.items').exists()).toBe(true);
   });
   it('renders correctly', () => {
    expect(createmeetup).toMatchSnapshot();
   });
  it('submits the form', () => {
    createmeetup.find('form').simulate('submit', {
      preventDefault: () => 'form'
    })
    expect(dispatch.mock.calls.length).toEqual(0)
  })
  it('updates state when input is changed', () => {
    createmeetup.find('input').first().prop('onChange')({
      target:{name: 'topic', value: 'Facebook for developers'}
    })
    expect(createmeetup.state('topic')).toEqual('Facebook for developers');
  });
})
