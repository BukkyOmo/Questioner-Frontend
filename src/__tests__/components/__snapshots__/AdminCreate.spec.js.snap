// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateMeetup /> renders correctly 1`] = `ShallowWrapper {}`;
